File: pet-service/src/main/java/com/phorus/petservice/repositories/PetRepository.java
package com.phorus.petservice.repositories;

import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;
import com.phorus.petservice.model.dbentities.Pet;


@Repository
public interface PetRepository extends JpaRepository<Pet, Long> {

    long countByUserId(Long userId);
}

File: pet-service/src/main/kotlin/com/phorus/petservice/repositories/PetRepository.kt
package com.phorus.petservice.repositories

import com.phorus.petservice.model.dbentities.Pet
import org.springframework.data.repository.kotlin.CoroutineCrudRepository
import org.springframework.stereotype.Repository

@Repository
interface PetRepository : CoroutineCrudRepository<Pet, Long> {

    suspend fun countByUserId(userId: Long): Long
}